core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitAdmin.java

```diff
@@ -112,13 +121,84 @@ public void createTable(
       throws ExecutionException {
     checkNamespace(namespace);
 
+    if (isTransactionMetadataDecouplingEnabled(options)) {
+      // For transaction metadata decoupling mode
+
+      throwIfTransactionMetadataDecouplingNotSupportedStorage(namespace);
+
+      String dataTableName = table + TRANSACTION_METADATA_DECOUPLING_DATA_TABLE_SUFFIX;
+      String txMetadataTableName = table + TRANSACTION_METADATA_DECOUPLING_METADATA_TABLE_SUFFIX;
+
+      if (admin.tableExists(namespace, dataTableName)) {
+        throw new IllegalArgumentException(
+            CoreError.TABLE_ALREADY_EXISTS.buildMessage(
+                ScalarDbUtils.getFullTableName(namespace, dataTableName)));
+      }
+      if (admin.tableExists(namespace, txMetadataTableName)) {
+        throw new IllegalArgumentException(
+            CoreError.TABLE_ALREADY_EXISTS.buildMessage(
+                ScalarDbUtils.getFullTableName(namespace, txMetadataTableName)));
+      }
+
+      // Create a data table
+      admin.createTable(namespace, dataTableName, metadata, options);
+
+      // Create a transaction metadata table
+      admin.createTable(
+          namespace,
```

Comment:
Just a clarification. Does this mean creating transaction metadata table in a different namespace is not supported?